x8.md

```diff
@@ -1,34 +1,38 @@
 ---
 sip: 3
-title: Initial Message Payload Specification Specification
+title: Initial Message Payload Specification
 status: Draft
 type: Standard
-author:
+author: Adam Babik <adam@status.im>, Oskar Thorén <oskar@status.im>
 created: 2019-04-22
 updated:
 ---
 
-**TODO: Remove everyting that doesn't touch on payload here**
-
-Using Adam's spec as starting point. Also be mindful of integration with the overview document.
-
-**TODO: Remove non transport privacy layer aspects here**
-
+- [Payload](#payload)
+- [Encoding](#encoding)
+  - [Future encoding format](#future-encoding-format)
+- [Content types](#content-types)
+- [Message types](#message-types)
+- [Clock vs Timestamp and message ordering](#clock-vs-timestamp-and-message-ordering)
+- [Replies](#replies)
+- [Upgradability](#upgradability)
 
-# Payload
+This specification describes how the payload of each message in the Status Protocol looks like. It does not care how the payload is encrypted or exchanged later.
 
 # Payload
 
-Payload is an array of bytes exchanged between peers. In the case of this protocol, the payload is encoded using [transit format](https://github.com/cognitect/transit-format). It is a text-based format so even encoded version is easily readable by humans. For more information about the payload in Status, please see [Initial Message Payload Specification Specification](x8.md).
-
-Payload contains the following fields:
+Payload is a struct (a compound data type) with the following fields:
```

Comment:
struct like In C or protobuf or what more precisely? I.e. if I want to implement a client in JS what do I need?

Reply:
Oops, nevermind. I got confused and forgot this was encoded in Transit, just didn't recognize the notion of structs from it :D